src/jit/lsraxarch.cpp

```diff
@@ -940,39 +942,49 @@ int LinearScan::BuildShiftRotate(GenTree* tree)
     {
         assert((source->OperGet() == GT_LONG) && source->isContained());
 
-        GenTree* sourceLo = source->gtOp.gtOp1;
-        GenTree* sourceHi = source->gtOp.gtOp2;
+        GenTree* sourceLo = source->gtGetOp1();
+        GenTree* sourceHi = source->gtGetOp2();
         assert(!sourceLo->isContained() && !sourceHi->isContained());
         RefPosition* sourceLoUse = BuildUse(sourceLo, srcCandidates);
         RefPosition* sourceHiUse = BuildUse(sourceHi, srcCandidates);
 
-        if (tree->OperGet() == GT_LSH_HI)
+        if (!tree->isContained())
```

Comment:
Hmm, when is a shift node contained?

Reply:
A shift node can be contained under a `STOREIND` when it is part of a memory RMW op. In most cases this is just handled in `BuildIndir`, but shift ops have special register requirements.